src/engine/TextIndexScanForEntity.cpp

```diff
@@ -139,6 +139,9 @@ string TextIndexScanForEntity::getCacheKeyImpl() const {
   os << "ENTITY INDEX SCAN FOR WORD: "
      << " with word: \"" << config_.word_ << "\" and fixed-entity: \""
      << (hasFixedEntity() ? fixedEntity() : "no fixed-entity") << " \"";
+  if (config_.scoreVar_.has_value()) {
+    os << " and scoreVar: \"" << config_.scoreVar_.value().name() << "\"";
+  }
```

Comment:
Please add a unit test for this in `test/engine/TextIndexScanForEntityTest.cpp`, there are already some other tests for cache keys.